### PR TITLE
Apply workaround for JDK-8329528

### PR DESCRIPTION
**Pull request by wendigo**
https://bugs.openjdk.org/browse/JDK-8329528

`G1NumCollectionsKeepPinned`: "After how many GCs a region has been found pinned G1 should  give up reclaiming it."

Fixes https://github.com/trinodb/trino/issues/21989

This flag defaults to 8: https://github.com/openjdk/jdk22u/blob/f5ce46f27e5489234d1305e0573e42898c9e9e32/src/hotspot/share/gc/g1/g1_globals.hpp#L327

More info:
https://github.com/openjdk/jdk22u/blob/f5ce46f27e5489234d1305e0573e42898c9e9e32/src/hotspot/share/gc/g1/g1CollectionSet.cpp#L370

### CODE REVIEW

**Review by findepi — APPROVED** [review]
🤞

**Review by findepi — COMMENTED** [review]
Update deployment.md

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeCompatibility.java [hunk 1]
```diff
@@ -68,18 +68,21 @@ public void extendEnvironment(Environment.Builder builder, Map<String, String> e
...
-                .withCopyFileToContainer(forHostPath(dockerFiles.getDockerFilesHostPath("conf/presto/etc/jvm.config")), containerConfigDir + "jvm.config")
+                .withCopyFileToContainer(forHostPath(dockerFiles.getDockerFilesHostPath(jvmConfig)), containerConfigDir + "jvm.config")
                 .withCopyFileToContainer(forHostPath(configDir.getPath(getConfigFileFor(dockerImage))), containerConfigDir + "config.properties")
                 .withCopyFileToContainer(forHostPath(configDir.getPath(getHiveConfigFor(dockerImage))), containerConfigDir + "catalog/hive.properties")
                 .withCopyFileToContainer(forHostPath(configDir.getPath("iceberg.properties")), containerConfigDir + "catalog/iceberg.properties")
                 .withCopyFileToContainer(forHostPath(dockerFiles.getDockerFilesHostPath()), "/docker/presto-product-tests")
                 .withStartupCheckStrategy(new IsRunningStartupCheckStrategy())
                 .waitingForAll(forLogMessage(".*======== SERVER STARTED ========.*", 1), forHealthcheck())
                 .withStartupTimeout(Duration.ofMinutes(5));
+
```
Comment (findepi) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeCompatibility.java:85]: revert

File: testing/trino-product-tests-launcher/src/main/resources/docker/presto-product-tests/conf/presto/etc/jvm-pre-jdk22.config [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+-Djdk.attach.allowAttachSelf=true
+# jdk.nio.maxCachedBufferSize controls what buffers can be allocated in per-thread "temporary buffer cache" (sun.nio.ch.Util). Value of 0 disables the cache.
+-Djdk.nio.maxCachedBufferSize=0
+-DHADOOP_USER_NAME=hive
+-Duser.timezone=Asia/Kathmandu
+-XX:ErrorFile=/docker/logs/product-tests-presto-jvm-error-file.log
+# Allow loading dynamic agent used by JOL
+-XX:+EnableDynamicAgentLoading
+-XX:+UnlockDiagnosticVMOptions
+# https://bugs.openjdk.org/browse/JDK-8329528
```
Comment (findepi) [inline testing/trino-product-tests-launcher/src/main/resources/docker/presto-product-tests/conf/presto/etc/jvm-pre-jdk22.config:20]: remove

File: .mvn/jvm.config [hunk 3]
```diff
@@ -11,3 +11,4 @@
 --add-opens=jdk.compiler/com.sun.tools.javac.comp=ALL-UNNAMED
 -XX:+UnlockDiagnosticVMOptions
 -XX:+ExitOnOutOfMemoryError
+-XX:G1NumCollectionsKeepPinned=10000000
```
Comment (losipiuk) [inline .mvn/jvm.config:14]: would that be wise to fail on startup if we see that JVM version is one without fix and we do not have `-XX:G1NumCollectionsKeepPinned=10000000` set?

**Review by losipiuk — APPROVED** [review]

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/util/Test.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+import org.apache.commons.compress.archivers.tar.TarArchiveEntry;
+import org.apache.commons.compress.archivers.tar.TarArchiveInputStream;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class Test
+{
```
[the remark this answers is not in the document]
Comment (wendigo) [reply testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/util/Test.java:25]: Damn. Garbage. Sorry.

**Review by losipiuk — APPROVED** [review]

**Pull request merged**